features/machine/machine.feature

```diff
@@ -141,4 +141,92 @@ Feature: Machine features testing
     Given I scale the machineset to +2
     Then the machineset should have expected number of running machines
 
-    
+  # @author zhsun@redhat.com
+  @admin
+  @destructive
+  Scenario Outline: Machineset should have relevant annotations to support scale from/to zero
+    Given I have an IPI deployment
+    And I switch to cluster admin pseudo user
+    And I use the "openshift-machine-api" project
+    And I pick a random machineset to scale
+
+    Given I run the :get admin command with:
+      | resource      | machineset              |
+      | resource_name | <%= machine_set.name %> |
+      | o             | yaml                    |
+    Then the step should succeed
+    And I save the output to file> machineset.yaml
+
+    # Create a machineset with a valid instanceType
+    And I replace content in "machineset.yaml":
+      | <%= machine_set.name %> | <machineset_name>-valid |
+      | <re_type_field>         | <valid_value>           |
+      | /replicas:.*/           | replicas: 1             |
+
+    When I run the :create admin command with:
+      | f | machineset.yaml |
+    Then the step should succeed
+    And admin ensures "<machineset_name>-valid" machineset is deleted after scenario
+
+    When I run the :annotate admin command with:
+      | resource     | machineset               |
+      | resourcename | <machineset_name>-valid  |
+      | overwrite    | true                     |
+      | keyval       | new=new                  |
+    Then the step should succeed
+
+    When I run the :describe admin command with:
+      | resource | machineset              |
+      | name     | <machineset_name>-valid |
+    Then the step should succeed
+    And the output should contain:
+      | machine.openshift.io/memoryMb: |
+      | machine.openshift.io/vCPU:     |
+      | new:                           |
+
+    # Create a machineset with an invalid instanceType
+    And I replace content in "machineset.yaml":
+      | <%= machine_set.name %> | <machineset_name>-invalid |
+      | <re_type_field>         | <invalid_value>           |
+      | /replicas:.*/           | replicas: 1               |
+
+    When I run the :create admin command with:
+      | f | machineset.yaml |
+    Then the step should succeed
+    And admin ensures "<machineset_name>-invalid" machineset is deleted after scenario
+
+    Given 1 pods become ready with labels:
+      | api=clusterapi,k8s-app=controller |
+    And I wait for the steps to pass:
+    """
+    When I run the :logs admin command with:
+      | resource_name | <%= pod.name %>    | 
+      | c             | machine-controller |
+    Then the step should succeed
+    And the output should contain:
+      | ReconcileError |
```

Comment:
What you are capturing, is a very generic error which could be caused by other machinesets. It's more decent to catch the output that explicitly reports which machineset has `ReconcileError`. 

For example this log: `I0318 06:52:29.312631       1 recorder.go:52] controller-runtime/manager/events "msg"="Warning"  "message"="unknown instance type: invalid" "object"={"kind":"MachineSet","namespace":"openshift-machine-api","name":"zhsun6-48nkt-worker-us-east-2c-invalid","uid":"e79de785-5c99-4aa0-8ae7-3d09b8514302","apiVersion":"machine.openshift.io/v1beta1","resourceVersion":"150701"} "reason"="ReconcileError"`

To capture the log, you could use
```
Then the output should match:
  | <%= machine_set.name %>.*ReconcileError |
```

Note I haven't validated the regexp myself, I'm just trying to show an example.